cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -372,7 +372,7 @@ namespace
                     {
                         BuiltinPtr b = dynamic_pointer_cast<Builtin>(p);
                         assert(b);
```

Comment:
The 2nd place where we used to `typeId`. But here we already know that we're dealing with a `builtin` type, so using the virtual function `typeId` was pointless anyways. I switched it to just use the `kindAsString` method which returns `int`, `string`, `Value`, etc. and is specific to built-in types.